crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -210,6 +210,75 @@ fn assert_component_access_compatibility(
            query_type, filter_type, system_name, accesses);
 }
 
+/// A collecton of potentially conflicting [`SystemParam`]s allowed by disjoint access.
+///
+/// Allows to safely access and interact with up to 8 mutually exclusive [`SystemParam`]s such as
+/// two queries that reference the same mutable data  or a mutable query and acess to the ecs world.
+///
+/// Access to each individual [`SystemParam`] by using the functions `p0()`, `p1()`, ..., `p7()`. This ensures that
+/// there's either only one mutable reference to a parameter at a time or any number of immutable references.
+///
+/// # Example
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// #
+/// # #[derive(Component)]
+/// # struct Health;
+/// #
+/// # #[derive(Component)]
+/// # struct Enemy;
+/// #
+/// # #[derive(Component)]
+/// # struct Ally;
+/// #
+/// // Given the following system
+/// fn fancy_system(
+///     mut set: ParamSet<(
+///         Query<&mut Health, With<Enemy>>,
+///         Query<&mut Health, With<Ally>>,
+///         &World
+///     )>
+/// ) {
+///     // To access the ParamSet's elements according to the order they're defined, use the methods p0(), p1(), ..., p7().
+///     for mut health in set.p0().iter_mut() {
+///         // Do your fancy stuff here...
+///     }
+///
+///     for mut health in set.p1().iter_mut() {
+///         // Do even fancier stuff here...
+///     }
+///
+///     let entities = set.p2().entities();
+///     // Note that you can only mutably access one parameter of the [`ParamSet`] at a time due to Rust's borrowing rules.
+/// }
+/// ```
+///
+/// # Panics
+///
+/// ```should_panic
+/// # use bevy_ecs::prelude::*;
+/// #
+/// # #[derive(Component)]
+/// # struct Health;
+/// #
+/// # #[derive(Component)]
+/// # struct Enemy;
+/// #
+/// # #[derive(Component)]
+/// # struct Ally;
+/// #
+/// // This is an example of a system that panics because of the existance of the mutually exclusive system params.
```

Comment:
```suggestion
/// // This is an example of a system that panics because of the existence of the mutually exclusive system params.
/// // An entity may have both the Enemy and Ally components, causing two mutable references to the same Health component to be active at once.
```